### PR TITLE
Add CI and docs

### PR DESCRIPTION
**Pull request by andreaTP**
Questions:
- should we include the license file in this repo? (pending ongoing discussion)
- anything else I missed from the checklist: https://github.com/dotnet-foundation/projects/issues/353#issue-2096112950 ?

Pending:
- how do we add the CLA?
- ~add the relevant PAT token to the secrets~ done

### CODE REVIEW

File: .github/workflows/dependabot-automerge.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v2.0.0
+      - name: Enable auto-merge for Dependabot PRs
+        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
```
Comment (baywet) [inline .github/workflows/dependabot-automerge.yaml]: I think minor could be included here as well

**Review by baywet — COMMENTED** [review]
To answer your questions:
- Yes I believe the license file should be included since this is "the actual source" of the code.
- The CLA will be an app/bot to install to the repo that will register hooks, and once we do, it'll be available as a check as well, as I believe it's the same we're using at Microsoft. I expect the DNF folks to give us the specifics/don't hesitate to ask on the other issue we have under DNF.

File: .github/workflows/dependabot-automerge.yaml (hunk 1 above)
Comment (andreaTP) [reply .github/workflows/dependabot-automerge.yaml]: done

**Comment by andreaTP** [comment]
Thanks! This is ready for review.

> don't hesitate to ask on the other issue we have under DNF

I'll do it as soon as we get the two PRs merged and we do a successful test release.

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**